cf-custom-resources/lib/wkld-custom-domain.js

```diff
@@ -0,0 +1,554 @@
+      // Wait up to 5 minutes
+      $waiter: {
+        delay: DELAY_RECORD_SETS_CHANGE_IN_S,
+        maxAttempts: ATTEMPTS_RECORD_SETS_CHANGE,
+      },
+      Id: ChangeInfo.Id,
+    })
+    .promise();
+}
+
+/**
+ *
+ * @param {Set<String>} aliases
+ * @param {String} publicAccessDNS
+ * @param {String} publicAccessHostedZoneID
+ * @returns {Promise<void>}
+ */
+async function deactivate(aliases, publicAccessDNS, publicAccessHostedZoneID) {
+  let promises = [];
+  for (let alias of aliases) {
+    promises.push(
+      deactivateAlias(alias, publicAccessDNS, publicAccessHostedZoneID)
+    );
+  }
+  await Promise.all(promises);
+}
+
+/**
+ * Remove the A-record of an alias that points to the load balancer DNS from its corresponding hosted zone.
+ *
+ * @param {String} alias
+ * @param {String} publicAccessDNS
+ * @param {String} publicAccessHostedZoneID
+ * @returns {Promise<void>}
+ */
+async function deactivateAlias(
+  alias,
+  publicAccessDNS,
+  publicAccessHostedZoneID
+) {
+  // NOTE: It has been validated that if the alias is in use, it is in use by the service itself.
+  // Therefore, an "UPSERT" will not overwrite a record that belongs to another service.
+  let changes = [
+    {
+      Action: "DELETE",
+      ResourceRecordSet: {
+        Name: alias,
+        Type: "A",
+        AliasTarget: {
+          DNSName: publicAccessDNS,
+          EvaluateTargetHealth: true,
+          HostedZoneId: publicAccessHostedZoneID,
+        },
+      },
+    },
+  ];
+
+  let { hostedZoneID, route53Client } = await domainResources(alias);
+  let changeResourceRecordSetsInput = {
+    ChangeBatch: {
+      Comment: `Delete the A-record for ${alias}`,
+      Changes: changes,
+    },
+    HostedZoneId: hostedZoneID,
+  };
+  let changeInfo;
+  try {
+    ({ ChangeInfo: changeInfo } = await route53Client
+      .changeResourceRecordSets(changeResourceRecordSetsInput)
+      .promise());
+  } catch (e) {
+    let recordSetNotFoundErrMessageRegex = new RegExp(
+      ".*Tried to delete resource record set.*but it was not found.*"
+    );
+    if (recordSetNotFoundErrMessageRegex.test(e.message)) {
+      return; // If we attempt to `DELETE` a record that doesn't exist, the job is already done, skip waiting.
+    }
+
+    let recordSetNotMatchedErrMessageRegex = new RegExp(
+      ".*Tried to delete resource record set.*but the values provided do not match the current values.*"
+    );
+    if (recordSetNotMatchedErrMessageRegex.test(e.message)) {
+      // NOTE: The alias target, or record value is not exactly what we provided
+      // E.g. the alias target DNS name is another load balancer
+      // This service should not delete the A-record if it is not being pointed to.
+      // However, this is an unexpected situation, we should log this information.
+      console.log(
+        `Received error when trying to delete A-record for ${alias}: ${e.message}. Perhaps the alias record isn't pointing to the load balancer ${publicAccessDNS}.`
```

Comment:
I can't really imagine how this error could happen after `validateAlias`, unless there is some super edge case going on. But since now this is used by both NLB and static site (fronted by CF), we can prolly remove the 
"load balancer" keyword.

Reply:
sounds good